src/views/SessionView/index.vue

```diff
@@ -168,16 +189,26 @@ export default {
       });
     },
     reconnect_attempt() {
-      this.sessionReconnecting = true;
+      this.$store.dispatch("user/sessionDisconnected");
+      if (!this.session || !this.session._id) {
+        this.showTroubleStartingModal();
+      }
     },
     connect() {
-      if (this.sessionReconnecting) {
-        if (this.session && this.session._id) {
-          // we still need to re-join the room after Socket.IO re-establishes the connection
+      this.$store.dispatch("user/sessionConnected");
+
+      if (this.session && this.session._id) {
+        if (
+          (!this.session.student ||
+            this.session.student._id !== this.user._id) &&
+          (!this.session.volunteer ||
+            this.session.volunteer._id !== this.user._id)
+        ) {
+          // join the session if we haven't done so already
           this.joinSession(this.session._id);
-        } else {
-          location.reload();
         }
```

Comment:
a way to avoid the crazy if statement could be:
```
if (this.session && this.session._id) {
  if (this.session.student && this.session.student._id === this.user._id) return location.reload();
  if (this.session.volunteer && this.session.volunteer._id === this.user.Id) return location.reload();
  return this.joinSession(this.session._id);
}
```

i think this looks more readable, but up to you whether you want to change or not